src/map/skill.cpp

```diff
@@ -7756,9 +7829,19 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 	case TR_MYSTIC_SYMPHONY:
 	case TR_KVASIR_SONATA:
 	case EM_SPELL_ENCHANTING:
+	case NPC_DAMAGE_HEAL:
 		clif_skill_nodamage(src,bl,skill_id,skill_lv,
 			sc_start(src,bl,type,100,skill_lv,skill_get_time(skill_id,skill_lv)));
 		break;
+	case NPC_GRADUAL_GRAVITY:
+		clif_skill_nodamage(src, bl, skill_id, skill_lv, 1);
+		status_change_start(src, bl, type, 10000, skill_lv, 0, 0, 0, skill_get_time(skill_id, skill_lv), SCSTART_NOAVOID|SCSTART_NOTICKDEF|SCSTART_NORATEDEF);
+		break;
+	case NPC_ALL_STAT_DOWN:
+		clif_skill_nodamage(src, bl, skill_id, skill_lv, 1);
+		status_change_start(src, bl, type, 10000, skill_lv, 0, 0, 0, skill_get_time(skill_id, skill_lv), SCSTART_NOAVOID|SCSTART_NOTICKDEF|SCSTART_NORATEDEF);
+		clif_skill_damage(src,bl,tick, status_get_amotion(src), 0, -30000, 1, skill_id, skill_lv, DMG_SINGLE);
```

Comment:
Why do we tell the client -30000 damage?

Reply:
![image](https://user-images.githubusercontent.com/35770095/210580421-049fe0d7-18bb-4ef6-b2dd-58287db3e898.png)
official